src/test/kotlin/device/mothers/DeviceMother.kt

```diff
@@ -8,6 +8,11 @@ object DeviceMother {
         id = IdentifierMother.getValidIdentifier(),
         name = DeviceNameMother.getValidDeviceName(),
     )
-    fun getIdPrimitiveFrom(device: Device) = device.id.id
-    fun getNamePrimitiveFrom(device: Device) = device.name.name
+
+    fun getDeviceWithDifferentNameFrom(device: Device) = device.copy(
```

Comment:
The copy isn't really needed if you change all the fields (like you are doing here). If you change all the fields a new instatiaton is less confusing and verbose